Sources/SwiftProtobuf/ProtoNameResolvers.swift

```diff
@@ -12,22 +12,20 @@
 /// This type contains helper functions to resolve field names based on their
 /// numbers during encoding.
 ///
-/// This type is public since it might be useful to clients who want to write
-/// advanced custom visitors.
-public enum ProtoNameResolvers {
+internal enum ProtoNameResolvers {
 
   /// Returns a function that resolves the proto/text name for fields defined on
   /// the given message or in any set extensions.
   ///
   /// If the name cannot be resolved (because the field number is not defined
   /// on the message or any of its extensions, or names were not compiled into
   /// the binary), then the resolver returns nil.
-  public static func protoFieldNameResolver(
+  static func protoFieldNameResolver(
     for message: Message
-  ) -> (Int) -> String? {
+  ) -> (Int) -> StaticString? {
```

Comment:
Changing to `StaticString` here is a noticeable performance bump since the JSON and Text encoders can now just copy the UTF8 bytes across.